presto-main/src/main/java/io/prestosql/operator/PagesHash.java

```diff
@@ -168,6 +186,63 @@ public int getAddressIndex(int rightPosition, Page hashChannelsPage, long rawHas
         return -1;
     }
 
+    /**
+     * return array passed as an argument for performance reasons
+     */
+    public void getAddressIndex(int firstRightPosition, int rightPositions, Page hashChannelsPage, long[] rawHashes, long[] result)
+    {
+        verify(rightPositions == rawHashes.length, "Number of positions must match number of hashes");
+
+        /*
+         * Instead of looking for a match on a per-position basis, we execute every step in a "vectorized" way,
+         * so that all cpu mechanisms (caching, out-of-order execution, etc.) are fully utilized.
+         */
+        int[] pos = new int[rightPositions];
```

Comment:
Do we actually need `pos` array? It seems we could just use result here

Reply:
I tried putting -2 in result array as an indicator that the value has been found instead of using pos array but that slowed down benchmarks.